core/src/test/java/org/elasticsearch/index/mapper/string/SimpleStringMappingTests.java

```diff
@@ -562,9 +562,52 @@ public void testBackwardCompatible() throws Exception {
                 .startObject("field1")
                 .field("type", "string")
                 .field("position_offset_gap", 10)
-                .endObject().endObject().endObject().endObject().string();
+.endObject().endObject().endObject().endObject().string();
         parser.parse(mapping);
 
         assertThat(parser.parse(mapping).mapping().toString(), containsString("\"position_increment_gap\":10"));
     }
+
+    /**
+     * Test backward compatibility when a search_analyzer is specified without
+     * an index_analyzer
+     */
+    public void testBackwardCompatibleSearchAnalyzerMigration() throws Exception {
+
+        Settings settings = Settings.settingsBuilder()
+                .put(IndexMetaData.SETTING_VERSION_CREATED, VersionUtils.randomVersionBetween(random(), Version.V_1_0_0, Version.V_1_7_1))
+                .build();
+
+        DocumentMapperParser parser = createIndex("backward_compatible_index", settings).mapperService().documentMapperParser();
+
+        String mapping = XContentFactory.jsonBuilder().startObject().startObject("type").startObject("properties").startObject("field1")
+                .field("type", "string").field("search_analyzer", "keyword").endObject().endObject().endObject().endObject().string();
+        parser.parse(mapping);
+
+        assertThat(parser.parse(mapping).mapping().toString(), containsString("\"search_analyzer\":\"keyword\""));
```

Comment:
Instead of relying on serialization, can the test look at the parsed mapping and see that the search analyzer is equal to keyword? And also check the index analyzer is the default?